.github/actions/deploy-cloud-run/action.yml

```diff
@@ -45,11 +66,143 @@ runs:
       env:
         IMAGE: ${{ inputs.image }}
         IMAGE_INTERNAL: ${{ inputs.image_internal }}
+    - name: Set Host header
+      id: set_header
+      if: ${{ inputs.hostname != '' }}
+      run: |
+        if [[ -n "${{ inputs.hostname }}" ]]; then
+          echo "customHeaders={\"Host\": \"${{ inputs.hostname }}\"}" >> "${GITHUB_OUTPUT}"
+        else
+          echo "customHeaders={}" >> "${GITHUB_OUTPUT}"
+        fi
+      shell: bash
+    - name: Get current Cloud Run service revision
+      id: get_revision
+      if: ${{ inputs.automatic_rollback_enabled == 'true' || inputs.blue_green_enabled == 'true' }}
+      shell: bash
+      run: |
+        echo "Getting current cloudrun revision..."
+        REV=$( \
+          gcloud run services describe "${{ inputs.service }}" \
+            --region "${{ inputs.region }}" \
+            --format="value(status.latestReadyRevisionName)" \
+        )
+        echo "Current revision: $REV"
+        echo "revision=$REV" >> "$GITHUB_OUTPUT"
+    - name: Logging - Deploy to Cloud Run
+      run: echo "Deploying to Cloud Run..."
+      if: ${{ inputs.blue_green_enabled == 'false' }}
+      shell: bash
     - name: Deploy to Cloud Run
       uses: google-github-actions/deploy-cloudrun@v2
+      if: ${{ inputs.blue_green_enabled == 'false' }}
+      id: deploy
+      with:
+        service: ${{ inputs.service }}
+        job: ${{ inputs.job }}
+        image: ${{ inputs.image_internal || inputs.image }}
+        region: ${{ inputs.region }}
+        revision_traffic: ${{ inputs.to_latest == 'true' && 'LATEST=100' || '' }}
+        skip_default_labels: true
+    - name: Run HTTP health check
+      continue-on-error: true
+      uses: fjogeleit/http-request-action@v1
+      if: ${{ inputs.blue_green_enabled == 'false' }}
+      id: test_live
+      with:
+        url: ${{ steps.deploy.outputs.url }}${{ inputs.healthcheck_path }}
```

Comment:
Cloud Run has a mechanism that automatically checks for communication and health and switches revisions, and using this mechanism would be more effective for normal health checks. This process is only necessary to perform health checks from an external URL against a server that behaves like a reverse proxy, such as rearth-cloud.

In light of this, I believe that the health check URL should be able to specify the full URL, not the path.

https://cloud.google.com/run/docs/configuring/healthchecks#tcp-startup-probe

Reply:
@rot1024 agree on this. For internal health checks we can rely on the available probes. Meaning the internal healthcheck mechanism will be controlled by module definiition on infrastructure repo.

For this LoC, I will change it is to simulate external party/client calling the URL.

To summarize:

1. Update this line to simulate calling the FQDN + Path from external client
2. TODO in infrastructure repo: observe if we can add startup and liveness probe on each cloud run service module

Reply:
@rot1024 but for the blue testing I have another opinion. Before promoting the blue service to green/live we need to ensure the blue can serve the traffic properly. 

In order to specifically accessing the blue instance only without exposing it to consumer traffic / 0% live traffic we need to access it using tag---url.

For testing the green and determine whether we can finish or need to rollback to previous version I agree that full real URL with path is the right choice